### PR TITLE
[BUG FIX] Handles edge case where requested current_page > total_pages.

### PR DESCRIPTION
**Pull request by f3z0**
#### Purpose
Fixes an edge case whereby the client requests a current_page > total_pages.

#### Changes
ActiveModelSerializers::Adapter::JsonApi::PaginationLinks#next_page_url should return nil if there is no next_page however if the client incorrectly requests a pager greater than the total number of pages it will return an invalid next_page. Also similarly if the current_page is greater than total pages the prev_page_url should return the last previous page with results which would be equivalent to last_page_url. 

#### Caveats
None.

#### Related GitHub issues
Did not see any.

#### Additional helpful information
N/A

### CODE REVIEW

**Comment by f3z0** [comment]
@bf4 should I be concerned with the failing tests, they seem unrelated to my changes?

**Comment by wasifhossain** [comment]
Hey @f3z0, could you please rebase your commit (https://github.com/rails-api/active_model_serializers/pull/2399/commits/df9b7a85f59282c1fb174e1a1914c740c699326c) on top of `0-10-stable` branch and push that commit here again. with that, I think Github Actions would be able to run the tests for your changes.

Thanks.

**Comment by f3z0** [comment]
@wasifhossain strange I squashed rebased and force pushed and github actions did not change, codedclimate did not re-run as far as I can tell. Should I close this PR and reopen? Appreciate you helping me on this 🙏.

**Comment by wasifhossain** [comment]
oh I am sorry in that case. let me do the reopening for you :)

**Comment by wasifhossain** [comment]
seems like it didn't help either. we might need to configure something in the ci file. will get back to you if we find something.

thanks for the PR :+1:

**Comment by wasifhossain** [comment]
@f3z0 I have made a small addition in the CI file, so hopefully tests will run now if you rebase your branch again. thanks

**Comment by f3z0** [comment]
@wasifhossain awesome that worked 🙏

**Comment by wasifhossain** [comment]
tests look fine! could you please add a line in the changelog documenting this. thanks

**Comment by f3z0** [comment]
@wasifhossain yes absolutely, I've added under v0.10.12 fixes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ Misc:
 Fixes:
 
 - [#2398](https://github.com/rails-api/active_model_serializers/pull/2398) Update rails dependency to < 6.2 (@ritikesh)
+- [#2399](https://github.com/rails-api/active_model_serializers/pull/2399) Handles edge case where requested current_page > total_pages (@f3z0)
```
Comment (wasifhossain) [inline CHANGELOG.md:18]: as 0.10.12 has already been published, I guess this change would fit under Unreleased -> Fixes section.

Comment (f3z0) [reply CHANGELOG.md]: Sorry about that, this makes more sense.

**Comment by wasifhossain** [comment]
looks good to me! let's merge this :raised_hands:

**Pull request merged**